internal/rhp/v4/rhp.go

```diff
@@ -70,8 +70,24 @@ func (c *Client) ReadSector(ctx context.Context, hk types.PublicKey, hostIP stri
 }
 
 // WriteSector writes a sector to the host.
-func (c *Client) WriteSector(ctx context.Context, prices rhp4.HostPrices, token rhp4.AccountToken, rl rhp.ReaderLen, duration uint64) (rhp.RPCWriteSectorResult, error) {
-	panic("not implemented")
+func (c *Client) WriteSector(ctx context.Context, hk types.PublicKey, hostIP string, contract rhp.ContractRevision, prices rhp4.HostPrices, sk types.PrivateKey, token rhp4.AccountToken, rl rhp.ReaderLen, length, duration uint64) (res1 rhp.RPCWriteSectorResult, res2 rhp.RPCAppendSectorsResult, _ error) {
+	err := c.tpool.withTransport(ctx, hk, hostIP, func(t rhp.TransportClient) (err error) {
+		res1, err = rhp.RPCWriteSector(ctx, t, prices, token, rl, length, duration)
```

Comment:
Every client method should only map to a single RPC.

Reply:
Do we want to get fancy on the caller side (uploader) and make sure we avoid dialing the host twice?

Reply:
I wouldn't. To some extend we already do that with our reference counting in the transport pool but eventually we will remove the append from the worker anyway.